views/admin/users.erb

```diff
@@ -139,6 +152,67 @@
 <% append_script_declaration("WDN.initializePlugin('datepickers');") %>
 
 <% append_script_declaration(%q[require(['jquery'], function($) {
+
+    function buildSearchFilterList() {
+         let search = [];
+         // pull the values that the search filters had when the page loaded
+         // prevents a bug where a user could change the search filters without pressing the search button and switch to a page that would no longer be valid
+         if ($('#hidden-first-name').val() != '') {
+             search.push('first_name=' + $('#hidden-first-name').val());
+         }
+         if ($('#hidden-last-name').val() != '') {
+             search.push('last_name=' + $('#hidden-last-name').val());
+         }
+         if ($('#hidden-email').val() != '') {
+             search.push('email=' + $('#hidden-email').val());
+         }
+         if ($('#hidden-expiration-date').val() != '') {
+             search.push('expiration_date=' + $('#hidden-expiration-date').val());
+         }
+         if ($('#hidden-expiration-date-operation').val() != '') {
+             search.push('expiration_date_operation=' + $('#hidden-expiration-date-operation').val());
+         }
+         if ($('#hidden-studio-status').val() != '') {
+             search.push('studio_status=' + $('#hidden-studio-status').val());
+         }
+         if ($('#hidden-tool-authorization').val() != '') {
+             search.push('selected_tool=' + $('#hidden-tool-authorization').val());
+         }
+         return search;
+     }
+
+     function sort(email,name,expiration_date) {
+        // in case we have search filters we need to preserve these
+        let search = buildSearchFilterList();
+        let new_url = "?";
+        if (name){
+            if ($('#hidden-sort-by-name').val() == "asc"){
+                new_url += "hidden_sort_by_name=desc";
```

Comment:
When renaming the parameters so they don't have "hidden" in them you will have to update those here when building the URL.  The hidden field ID names make sense (so dont change those), but I just don't think it's necessary for the parameters that will be passed in the URL to have "hidden" in the name.